qa/hooks/scripts/qa-plan-stamp.sh

```
@@ -191,6 +309,234 @@ case "$VERB" in
+    # plugin that the runtime loads matches this source tree.
+    echo "qa-plan doctor"
+    echo "  repo:    $(git rev-parse --show-toplevel 2>/dev/null || echo '?')"
+    echo "  branch:  ${BRANCH:-<detached>}"
+    echo "  git dir: $GITDIR"
+    echo
+
+    _STAMP_JSON=$(cat "$STAMP" 2>/dev/null || echo "")
+    _V=$(qpg_stamp_valid "$_STAMP_JSON" "$BRANCH")
+    echo "  stamp:   $([ -n "$_STAMP_JSON" ] && echo "present" || echo "absent") [$_V]"
+    if [ -n "$_STAMP_JSON" ]; then
+      echo "           source=$(printf '%s' "$_STAMP_JSON" | jq -r '.approval_source // "<none>"' 2>/dev/null || echo '?')"            "approver=$(printf '%s' "$_STAMP_JSON" | jq -r '.approver // "<none>"' 2>/dev/null || echo '?')"
```

Shellcheck (0.11.0)

[info] Quote repeated spaces to avoid them collapsing into one.

(SC2291)